### PR TITLE
roachtest: upgrade GCE to n2 and AWS to m6id and c6id

### PR DESCRIPTION
**Pull request by srosenberg**
Previously, roachtest used n1 in GCE, m5d and c6d in AWS. CockroachDB Cloud hardware now uses n2 in GCE, m6i in AWS [1]. This change brings roachtest hardware into parity with CockroachDB Cloud; it is based on the draft PR in [2].

[1] https://cockroachlabs.atlassian.net/wiki/spaces/MC/pages/2799501550/CockroachDB+Cloud+Hardware
[2] https://github.com/cockroachdb/cockroach/pull/99991

Epic: none

Release note: None

### CODE REVIEW

**Review by erikgrinaker — APPROVED** [review]
Nice. Probably goes without saying that  we'll want an annotation for this, as it's going to throw our benchmarks out of whack.

**Review by herkolategan — APPROVED** [review]

**Review by tbg — APPROVED** [review]
Thanks! Also worth doing a canary run of 1-2 roachtests for both GCE and AWS, disregard if you've done it already.

**Comment by srosenberg** [comment]
> Also worth doing a canary run of 1-2 roachtests for both GCE and AWS, disregard if you've done it already.

Absolutely! Canary runs are queued up; won't merge until we have the signal.

**Comment by srosenberg** [comment]
AWS run succeeded, but GCE run had a couple of issues,

```
grep -B 6 "test failed" test_runner-1686184082.log|grep "Quota 'N2_CPUS' exceeded" |sort |uniq -c
   9  - Quota 'N2_CPUS' exceeded.  Limit: 24.0 in region europe-west2.
```

```
grep -B 5 "test failed" test_runner-1686184082.log |grep "Number of local"|sort |uniq -c
  39  - Number of local SSDs for an instance of type n2-highcpu-32 should be one of [0, 4, 8, 16, 24], while [1] is requested.: exit status 1.
   3  - Number of local SSDs for an instance of type n2-highcpu-96 should be one of [0, 16, 24], while [1] is requested.: exit status 1.
```

**Comment by srosenberg** [comment]
After further tweaks, it looks like everything passed,

- GCE: https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/10477812?buildTab=tests
- AWS: https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyAwsBazel/10477811?buildTab=tests

In the GCE run, there were two cluster creation errors, one of which is a known issue (resolved by [1]),

```
16:18:27 test_runner.go:668: [w13] Unable to create (or reuse) cluster for test restore/tpce/400GB/aws/nodes=9/cpus=8/zones=us-east-2b,us-west-2b,eu-west-1b due to: in provider: gce: Command: gcloud [compute instances create --subnet default --scopes cloud-platform --image ubuntu-2004-focal-v20210603 --image-project ubuntu-os-cloud --boot-disk-type pd-ssd --service-account 21965078311-compute@developer.gserviceaccount.com --maintenance-policy MIGRATE --create-disk type=,size=1000GB,auto-delete=yes --machine-type n2-standard-8 --labels usage=roachtest,cluster=teamcity-10477812-1686369260-155-n9cpu8-geo,lifetime=12h0m0s,arch=amd64,created=2023-06-10t16_17_46z,roachprod=true --metadata-from-file startup-script=/tmp/gce-startup-script3417406819 --project cockroach-ephemeral --boot-disk-size=32GB --zone us-east-2b teamcity-10477812-1686369260-155-n9cpu8-geo-0001 teamcity-10477812-1686369260-155-n9cpu8-geo-0002 teamcity-10477812-1686369260-155-n9cpu8-geo-0003]
Output: ERROR: (gcloud.compute.instances.create) Could not fetch resource:
```

the other one is a small quota in `europe-west4`, which I missed in the previous attempt,

```
15:28:26 test_runner.go:668: [w4] Unable to create (or reuse) cluster for test import/tpcc/warehouses=4000/geo due to: in provider: gce: Command: gcloud [compute instances create --subnet default --scopes cloud-platform --image ubuntu-2004-focal-v20210603 --image-project ubuntu-os-cloud --boot-disk-type pd-ssd --service-account 21965078311-compute@developer.gserviceaccount.com --maintenance-policy MIGRATE --local-ssd interface=NVME --local-ssd interface=NVME --machine-type n2-standard-16 --labels usage=roachtest,cluster=teamcity-10477812-1686369260-131-n8cpu16-geo,lifetime=12h0m0s,arch=amd64,created=2023-06-10t15_25_32z,roachprod=true --metadata-from-file startup-script=/tmp/gce-startup-script3945224579 --project cockroach-ephemeral --boot-disk-size=32GB --zone europe-west4-b teamcity-10477812-1686369260-131-n8cpu16-geo-0003 teamcity-10477812-1686369260-131-n8cpu16-geo-0004]
Output: Created [https://www.googleapis.com/compute/v1/projects/cockroach-ephemeral/zones/europe-west4-b/instances/teamcity-10477812-1686369260-131-n8cpu16-geo-0003].
WARNING: Some requests generated warnings:
 - Disk size: '32 GB' is larger than image size: '10 GB'. You might need to resize the root repartition manually if the operating system does not support automatic resizing. See https://cloud.google.com/compute/docs/disks/add-persistent-disk#resize_pd for details.
 - The resource 'projects/ubuntu-os-cloud/global/images/ubuntu-2004-focal-v20210603' is deprecated. A suggested replacement is 'projects/ubuntu-os-cloud/global/images/ubuntu-2004-focal-v20230605'.
ERROR: (gcloud.compute.instances.create) Could not fetch resource:
 - Quota 'N2_CPUS' exceeded.  Limit: 24.0 in region europe-west4.
	metric name = compute.googleapis.com/n2_cpus
	limit name = N2-CPUS-per-project-region
	dimensions = region: europe-west4
```

**Comment by srosenberg** [comment]
Quota in `europe-west4` has been lifted. Final run 🤞,

- GCE: https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/10484150?buildTab=tests
- AWS: https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyAwsBazel/10484149?buildTab=tests

**Comment by srosenberg** [comment]
Both runs succeeded, modulo the know issue with `restore/tpce/400GB/aws/nodes=9/cpus=8/zones=us-east-2b,us-west-2b,eu-west-1b` (in GCE). It's instructive to compare durations after and before this change,

- AWS ~7h 27m (3-run average on PR branch) vs. ~7h 46m (7-run average on master)
- GCE ~16h 42m (3-run average on PR branch) vs. ~19h 4m (7-run average on master)

N.B. variance between runs is much higher on master. Further study is queued up once we have more data points for this change.

TL;DR: shaving ~3 hours from a GCE run is a quick win, considering we were approaching 24h for some GCE runs.

**Comment by srosenberg** [comment]
TFTR!

bors r=erikgrinaker,tbg,herkolategan
